### DISCUSSION TITLE
NavigationViewItem ContentTemplate

### DISCUSSION THREAD

**Discussion by EtherGhost** [amwx/FluentAvalonia#272] [Q&A]
Hi!

How do you set the NavigationViewItem ContentTemplate?

**Comment by amwx** [comment]
`NavigationViewItem` is just a `ContentControl` so it works like any other.

If your items are bound, bind the VM into Content and set the ContentTemplate then:
```Xaml
<ui:NavigationView>
    <ui:NavigationView.MenuItemTemplate>
        <DataTemplate DataType="...">
            <!-- or declare the template inline like this one -->
            <ui:NavigationViewItem Content="{Binding }" ContentTemplate="{StaticResource MyDataTemplate}" />
        </DataTemplate>
    </ui:NavigationView.MenuItemTemplate>
<ui:NavigationView>
```

**Reply by EtherGhost** [reply]
I get "Resource is not found". What am I missing? I have removed unrelated stuff. 

```<UserControl.Resources>
        <DataTemplate x:Key="NaviItemTemplate" DataType="viewModels:NavigationItemViewModel">
            <controls:NavigationViewItem Content="{CompiledBinding}"
                                         ContentTemplate="{StaticResource NaviItemContentTemplate}"
                                         MenuItems="{CompiledBinding Items}"
                                         FontSize="16" />
        </DataTemplate>

        <DataTemplate x:Key="NaviItemContentTemplate" DataType="viewModels:NavigationItemViewModel">
            <StackPanel>
                <Button Content="{CompiledBinding DisplayMember}" />
            </StackPanel>
        </DataTemplate>
    </UserControl.Resources>

    <controls:NavigationView MenuItems="{CompiledBinding MenuItems}"
                             PaneDisplayMode="Left"
                             x:DataType="viewModels:NavigationViewModel"
                             MenuItemTemplate="{StaticResource NaviItemTemplate}" />```


![Screenshot from 2023-01-22 14-06-39](https://user-images.githubusercontent.com/40342638/213917589-6c2bec8f-b736-4dec-b9a0-b2bc12b1f341.png)

**Reply by EtherGhost** [reply]
![Screenshot from 2023-01-22 14-06-39](https://user-images.githubusercontent.com/40342638/213917633-618eb977-9497-46eb-9cd7-34638c397611.png)

**Reply by EtherGhost** [reply]
I was looking in to this because I'm also missing a "Command" property on that "NavigationViewItem". But I can get around that with something like this:

```<UserControl.Resources>
        <DataTemplate x:Key="NaviItemTemplate" DataType="viewModels:NavigationItemViewModel">
            <controls:NavigationViewItem Content="{CompiledBinding DisplayMember}"
                                         MenuItems="{CompiledBinding Items}"
                                         IsEnabled="{CompiledBinding !IsBusy}"
                                         FontSize="16">
                <i:Interaction.Behaviors>
                    <ia:EventTriggerBehavior EventName="Tapped">
                        <ia:InvokeCommandAction Command="{CompiledBinding OpenDetailViewCommand}" />
                    </ia:EventTriggerBehavior>
                </i:Interaction.Behaviors>
            </controls:NavigationViewItem>
        </DataTemplate>
    </UserControl.Resources>

    <controls:NavigationView MenuItems="{CompiledBinding MenuItems}"
                             MenuItemTemplate="{StaticResource NaviItemTemplate}"
                             PaneDisplayMode="Left"
                             x:DataType="viewModels:NavigationViewModel" />

**Reply by amwx** [reply]
> I get "Resource is not found". What am I missing? I have removed unrelated stuff.
> 
> ```
>         <DataTemplate x:Key="NaviItemTemplate" DataType="viewModels:NavigationItemViewModel">
>             <controls:NavigationViewItem Content="{CompiledBinding}"
>                                          ContentTemplate="{StaticResource NaviItemContentTemplate}"
>                                          MenuItems="{CompiledBinding Items}"
>                                          FontSize="16" />
>         </DataTemplate>
> 
>         <DataTemplate x:Key="NaviItemContentTemplate" DataType="viewModels:NavigationItemViewModel">
>             <StackPanel>
>                 <Button Content="{CompiledBinding DisplayMember}" />
>             </StackPanel>
>         </DataTemplate>
>     </UserControl.Resources>
> 
>     <controls:NavigationView MenuItems="{CompiledBinding MenuItems}"
>                              PaneDisplayMode="Left"
>                              x:DataType="viewModels:NavigationViewModel"
>                              MenuItemTemplate="{StaticResource NaviItemTemplate}" />```
> 
> 
> ![Screenshot from 2023-01-22 14-06-39](https://user-images.githubusercontent.com/40342638/213917589-6c2bec8f-b736-4dec-b9a0-b2bc12b1f341.png)
> ```

Swap the order of the DataTemplates. Resource dictionaries are ordered (and searched bottom up) so resource look-ups will fail if you try to use a resource declared further down in the dictionary

**Reply by EtherGhost** [reply]
Got it.
Thanks!

Really nice work by the way!
